packages/squiggle-lang/__tests__/Distributions/Symbolic_test.res

```diff
@@ -0,0 +1,161 @@
+open Jest
+open Expect
+open TestHelpers
+
+// TODO: use Normal.make (but preferably after teh new validation dispatch is in)
+let mkNormal = (mean, stdev) => GenericDist_Types.Symbolic(#Normal({mean: mean, stdev: stdev}))
+
+describe("(Symbolic) normalize", () => {
+  testAll("has no impact on normal distributions", list{-1e8, -1e-2, 0.0, 1e-4, 1e16}, mean => {
+    let normalValue = mkNormal(mean, 2.0)
+    let normalizedValue = run(FromDist(ToDist(Normalize), normalValue))
+    normalizedValue 
+    -> unpackDist
+    -> expect
+    -> toEqual(normalValue)
+  })
+})
+
+describe("(Symbolic) mean", () => {
+  testAll("of normal distributions", list{-1e8, -16.0, -1e-2, 0.0, 1e-4, 32.0, 1e16}, mean => {
+    run(FromDist(ToFloat(#Mean), mkNormal(mean, 4.0))) 
+    -> unpackFloat 
+    -> expect 
+    -> toBeCloseTo(mean)
+  })
+
+  Skip.test("of normal(0, -1) (it NaNs out)", () => {
+    run(FromDist(ToFloat(#Mean), mkNormal(1e1, -1e0)))
+    -> unpackFloat
+    -> expect
+    -> ExpectJs.toBeFalsy
+  })
+
+  test("of normal(0, 1e-8) (it doesn't freak out at tiny stdev)", () => {
+    run(FromDist(ToFloat(#Mean), mkNormal(0.0, 1e-8)))
+    -> unpackFloat
+    -> expect
+    -> toBeCloseTo(0.0)
+  })
+
+  testAll("of exponential distributions", list{1e-7, 2.0, 10.0, 100.0}, rate => {
+    let meanValue = run(FromDist(ToFloat(#Mean), GenericDist_Types.Symbolic(#Exponential({rate: rate}))))
+    meanValue -> unpackFloat -> expect -> toBeCloseTo(1.0 /. rate)  // https://en.wikipedia.org/wiki/Exponential_distribution#Mean,_variance,_moments,_and_median
+  })
+
+  test("of a cauchy distribution", () => {
+    let meanValue = run(FromDist(ToFloat(#Mean), GenericDist_Types.Symbolic(#Cauchy({local: 1.0, scale: 1.0}))))
+    meanValue
+    -> unpackFloat
+    -> expect
+    -> toBeCloseTo(2.01868297874546)
+    //-> toBe(GenDistError(Other("Cauchy distributions may have no mean value.")))
```

Comment:
Is this a mistake? (Meant to be removed)

Reply:
I wanted to leave the comment in for worlds where we get a proper dispatch set up that doesn't assume we want the non-analytical mean of `Cauchy`.